fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -25,179 +40,112 @@ NONE_METHOD_OPT="NONE"
 
-  [--slicetimerparams=<"@" separated list of slicetimer parameters>]
+# opts_AddOptional '--printcom' 'RUN' 'print-command' "DO NOT USE THIS! IT IS NOT IMPLEMENTED!"
+# Disable RUN
+RUN=""
 
-      Enables passing additional parameters to FSL's 'slicetimer' if --doslicetime="TRUE".
-      The parameters to pass should be provided as a "@" separated string, e.g.:
-        --slicetimerparams="--odd@--ocustom=<CustomInterleaveFile>"
-      For details about valid parameters please consult FSL's 'slicetimer' documentation. 
+opts_ParseArguments "$@"
 
-  [--fmrimask=<type of final mask to use for final fMRI output volume>]
-
-      Specifies the type of final mask to apply to the volumetric fMRI data. Valid options are:
-        "T1_fMRI_FOV" (default) - T1w brain based mask combined with fMRI FOV mask
-        "T1_DILATED_fMRI_FOV" - once dilated T1w brain based mask combined with fMRI FOV
-        "T1_DILATED2x_fMRI_FOV" - twice dilated T1w brain based mask combined with fMRI FOV
-        "fMRI_FOV" - fMRI FOV mask only (i.e., voxels having spatial coverage at all time points)
-      Note that mask is used in IntensityNormalization.sh, so the mask type affects the final results.
-
-  [--fmriref=<"NONE" (default) or reference fMRI run name>]
-
-      Specifies whether to use another (already processed) fMRI run as a reference for processing.
-      (i.e., --fmriname from the run to be used as *reference*).
-      The specified run will be used as a reference for motion correction and its distortion
-      correction and atlas (MNI152) registration will be copied over and used. The reference fMRI
-      has to have been fully processed using the fMRIVolume pipeline, so that a distortion
-      correction and atlas (MNI152) registration solution for the reference fMRI already exists. 
-      The reference fMRI must have been acquired using the same imaging parameters (e.g., phase 
-      encoding polarity and echo spacing), or it can not serve as a valid reference. (NO checking 
-      is performed to verify this).
-      WARNING: This option excludes the use of the --fmriscout option, as the scout from the
-      specified reference fMRI run is used instead. 
-      Please run with --processing-mode-info flag for additional information on the issues related 
-      to the use of --fmriref.
-
-  [--fmrirefreg=<"linear" (default) or "nonlinear">]
-
-      Specifies whether to compute and apply a nonlinear transform to align the inputfMRI to the 
-      reference fMRI, if one is specified using --fmriref. The nonlinear transform is computed 
-      using 'fnirt' following the motion correction using the mean motion corrected fMRI image.
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
 
-EOF
-}
+opts_ShowValues
 
 show_processing_mode_info() {
```

Comment:
@glasserm @mharms it isn't clear what to do with this "extra explanation" text.  `--help` is virtually the only flag that can't be intuitively transformed to take an argument (e.g., `--nocrop` can be changed to to `--crop=<yes/no>`), so newopts doesn't currently support 0-argument flags (`--help` is hardcoded into the parser).  This text used to be shown by another no-argument flag, which would take some extra code to support.

Do we convert it to a comment, add it to the --processing-mode description, add it to the legacy style error or warning message, or add special parsing code just to support a second help-like flag in this one script?

Reply:
I would think we would show that info when processing in legacy mode or when hcp-style is selected with non-compliant options.